### ISSUE TITLE
Effective Date for laws

### ISSUE THREAD

**Issue by BlazeGuy2** [StrangeLoopGames/EcoSuggestions#617]
_From @D3nnis3n on October 25, 2018 11:25_

Yes, i know, this is a feature request. But not having this feature has an effect that could be considered bad game balance and can fuck up servers quite much.

You have a tax law. You want to change it. Someone proposes a change. Unfortunatley, he proposes the change just a little after the old tax did it's taxing. The new law passes. Now you get taxed twice, which, depending of the content of the laws can destroy the economy.

It would be insaneley good if you could set an effective date for every law when it has passed.
For example:

The old tax law went in place at day 3, 2:00. The new one is proposed day 4 2:15.
To evade the aforementioned problem i set the law to come into effect on day 6, 0:00.
The law will not be in effect immediatly after voting - but it will evade the double-taxation.

This feature can be useful for MANY other things, too.

Now think about you changing the tax laws quite frequently to adapt to things happening and get double taxed in only half of the occasions. ;)

_Copied from original issue: StrangeLoopGames/EcoIssues#9861_

**Comment by BlazeGuy2** [comment]
_From @Deantwo on October 25, 2018 12:22_

So you are kinda asking for the ability to set a start time for periodic laws, from which it will loop from.
It would indeed be nice if we could do that, so we can set periodic tax laws to always trigger at midnight, no matter what time the law was passed.

On that note a small suggestion about periodic laws, can they be changed to have hour intervals instead of days? Default it to 24 hours of course, and maybe set a minimum interval.
Currently we have to set the periodic interval to `0.042` to have anything related to an hour, this is confusing and hard to use.
Might make a separate issue for this...

**Comment by BlazeGuy2** [comment]
_From @D3nnis3n on October 25, 2018 12:25_

Excactly. And yes, that would be a good idea too.

**Comment by BlazeGuy2** [comment]
_From @Deantwo on October 25, 2018 12:25_

@D3nnis3n, it is currently possible to make a "don't take effect until after this day" clause in a law. It is however a little annoying to look at.

Example:
```
Repeat every 1 day(s)
    If (current day > 6)
        Tax player...
    Otherwise
        Do nothing
```

**Comment by BlazeGuy2** [comment]
_From @D3nnis3n on October 25, 2018 12:27_

Mh, well if it's in this issue is pretty much useless. You can even do it per clause, so this should be fine. Meh. Can get very ugly with many if statements already.

I'd rather would like to have that as an option for a law though, as i don't really want to have these clauses in for the display in the webinterface or game. But this then obviously is a suggestion only and so i will move it there.

**Comment by BlazeGuy2** [comment]
_From @Deantwo on October 25, 2018 12:31_

Nah, don't close it. Like I said, I would love a way to set a "start time of day" for periodic laws.

**Comment by BlazeGuy2** [comment]
_From @D3nnis3n on October 25, 2018 12:32_

Reopen, shall the developers decide what to do with it.

**Comment by Deantwo** [comment]
So implemented in v0.9.0.0? :open_mouth:
